### PR TITLE
Fix dynamic class id validation

### PR DESCRIPTION
**Pull request by nut-neek**

### CODE REVIEW

**Review by MikePSQR — APPROVED** [review]
looks good

File: ran-core/src/main/java/io/ran/DynamicClassIdentifier.java [hunk 1]
```diff
@@ -42,8 +44,8 @@ public int hashCode() {
 	}
 
 	public static DynamicClassIdentifier create(String identifier) {
-		if (identifier == null || identifier.matches("[^0-9a-zA-Z_]")) {
-			throw new RuntimeException("Invalid dynamic class identifier. Only alphanumeric and _ is allowed in identifier");
+		if (!SourceVersion.isName(identifier)) {
+			throw new IllegalArgumentException("Invalid dynamic class identifier. It should be a valid JVM qualified name");
```
Comment (MikePSQR) [inline ran-core/src/main/java/io/ran/DynamicClassIdentifier.java:48]: maybe we want to include the identifier in the message?

**Comment by MikePSQR** [comment]
is single underscore ok?

**Comment by nut-neek** [comment]
> is single underscore supposed to be valid identifier?

Depends on a version of Java it seems

**Comment by MikePSQR** [comment]
> 

but do we want to manually forbid it always?

**Pull request merged**